### PR TITLE
Interpolate JSX tags inside translations

### PR DESCRIPTION
**Pull request by g3offrey**
Add the ability to interpolate JSX inside translations.

### CODE REVIEW

File: packages/react-i18n/src/utils/i18n.utils.js [hunk 1]
```diff
@@ -47,10 +47,43 @@ export const translate = (lang, i18nNames = {}) => {
...
+  const [_, beforeTagSiblings, component, children, afterTagSiblings] = regexResult;
+  const result = [];
+
+  if (beforeTagSiblings) {
+    result.push(...instantiateJSXInsideTranslation(beforeTagSiblings, renderers));
+  }
+
+  if (children) {
+    const parsedChildren = instantiateJSXInsideTranslation(children, renderers);
+    const renderResult = renderers[component]({ children: parsedChildren });
```
Comment (flepretre) [inline packages/react-i18n/src/utils/i18n.utils.js]: Not sure it works with classes 🤔 
```suggestion
    const renderResult = React.createElement(renderers[component], { children: parsedChildren });
```

Comment (g3offrey) [reply packages/react-i18n/src/utils/i18n.utils.js]: Indeed, you right. I live a bit in the future ! 😄 
Changing that 👍

File: packages/react-i18n/src/utils/__tests__/i18n.utils.spec.js [hunk 2]
```diff
@@ -159,6 +161,120 @@ describe('i18n translate function', () => {
       );
     });
   });
+
+  describe('with JSX', () => {
+    // eslint-disable-next-line react/prop-types
+    const Bold = ({ children }) => <strong>{children}</strong>;
+    // eslint-disable-next-line react/prop-types
```
Comment (flepretre) [inline packages/react-i18n/src/utils/__tests__/i18n.utils.spec.js]: You can disable for the whole file, we are in a test context

File: packages/react-i18n/src/utils/i18n.utils.js [hunk 3]
```diff
@@ -35,10 +36,90 @@ const pluralizeFunctions = {
...
+  }
+
+  return translationChildren;
+};
+
 export const translate = (lang, i18nNames = {}) => {
   const pluralize = pluralizeFunctions[_.get(lang, '_i18n.lang')] || pluralizeFunctions.fr;
 
-  return (key, data = {}, number, general) => {
+  return (key, data = {}, number, general, renderers) => {
```
Comment (mathieumure) [inline packages/react-i18n/src/utils/i18n.utils.js]: 🤔 If we made a major release, can we group in a config object ?
```suggestion
  return (key, { data = {}, number, general, renderers } = {}) => {
```

Comment (g3offrey) [reply packages/react-i18n/src/utils/i18n.utils.js]: Sure, I agree 😄 
However I prefer to do that in another PR released as major 👍

**Comment by g3offrey** [comment]
Thanks, I added a test case to show you what it does. 😄

**Pull request merged**